server/balancer.go

```diff
@@ -224,10 +224,25 @@ func (cb *capacityBalancer) Balance(cluster *clusterInfo) (*balanceOperator, err
 		followerPeers[storeID] = peer
 	}
 
+	var err error
+	removePeer := oldLeader
+	doLeaderBalance := true
 	newLeader := cb.selectNewLeaderPeer(cluster, followerPeers)
 	if newLeader == nil {
```

Comment:
If there is no leader on this store, you will get nothing at line 138. So this will not work.